### DISCUSSION TITLE
Playlist.removeTracks совершает много запросов

### DISCUSSION THREAD

**Discussion by amd64fox** [Chimildic/goofy#77] [Помощь с ошибкой] [answered]
При тесте новой функции удалял из пробного плейлиста около 11 треков, удивило кардинальное различие по времени исполнения скрипта в сравнении с `Playlist.saveWithUpdate`

Новая функция `Playlist.removeTracks`
```javascript
  let tracks_old = Cache.read('History.json');

  Playlist.removeTracks('0lGLC1MmRSwvkOj3Zg3g2r', tracks_old);
```
Итог: время работы скрипта 69 сек.

Старый метод удаления через  `Playlist.saveWithUpdate`, 
```javascript
    let tracks_old = Cache.read('History.json');
    let test = Source.getPlaylistTracks('test', '0lGLC1MmRSwvkOj3Zg3g2r');
    Filter.removeTracks(test, tracks_old);
    
    Playlist.saveWithUpdate({
       id: '0lGLC1MmRSwvkOj3Zg3g2r', 
       tracks: test,
       });
```
Итог:  10 сек.  

Сразу уточню, кэш `History.json` = ~ 45 тысяч. + плейлист `test` имеет ~1600 

И еще инфа по запросам:
`Playlist.removeTracks` = 445
`Playlist.saveWithUpdate` = 36

**Comment by Chimildic** [comment, marked as answer]
Внутри `Playlist.saveWithUpdate` треки фильтруются, чтобы отобрать только существующие.

`Playlist.removeTracks` просто пробрасывает все треки. За один запрос 100 треков. При 45 тысячах как раз выйдет 445 запросов. 

Не предполагал, что возникнет такая ситуация. Поправил.
